src/cats/protocols.cljc

```diff
@@ -97,3 +97,9 @@
 (defprotocol MonadTrans
   "A monad transformer abstraction."
   (-lift [m mv] "Lift a value from the parameterized monad to the transformer."))
+
+;; More info: http://en.wikipedia.org/wiki/Semilattice
+
+(defprotocol JoinSemiLattice
+  "A algebraic structure with element joins."
+  (-join [_ other] "Converge two data structures."))
```

Comment:
Or rename join to mjoin xD, or just do not have any public api on cats.core for semilattice...